src/commands/logs.ts

```diff
@@ -0,0 +1,354 @@
+import fs from 'fs';
+import path from 'path';
+
+import chalk from 'chalk';
+import dedent from 'dedent';
+import type { Command } from 'commander';
+
+import cliState from '../cliState';
+import logger from '../logger';
+import { wrapTable } from '../table';
+import { printBorder } from '../util/index';
+import { findLogFile, formatFileSize, getLogDirectory, getLogFiles } from '../util/logs';
+
+type LogType = 'debug' | 'error' | 'all';
+
+/**
+ * Applies syntax highlighting to log lines based on log level
+ */
+function highlightLogLines(lines: string[], noColor: boolean): string {
+  if (noColor) {
+    return lines.join('\n');
+  }
+
+  return lines
+    .map((line) => {
+      if (line.includes('[ERROR]')) {
+        return chalk.red(line);
+      } else if (line.includes('[WARN]')) {
+        return chalk.yellow(line);
+      } else if (line.includes('[DEBUG]')) {
+        return chalk.cyan(line);
+      } else if (line.includes('[INFO]')) {
+        return chalk.white(line);
+      }
+      return line;
+    })
+    .join('\n');
+}
+
+/**
+ * Prints a header with file information
+ */
+function printLogHeader(logPath: string, isCurrentSession: boolean): void {
+  const stats = fs.statSync(logPath);
+
+  printBorder();
+  logger.info(chalk.bold(path.basename(logPath)));
+  logger.info(chalk.gray(`Path: ${logPath}`));
+  logger.info(chalk.gray(`Size: ${formatFileSize(stats.size)}`));
+  logger.info(chalk.gray(`Modified: ${stats.mtime.toLocaleString()}`));
+
+  if (isCurrentSession) {
+    logger.info(chalk.yellow('Note: This is the log file for the current CLI session.'));
+  }
+
+  printBorder();
+}
+
+interface PrintOptions {
+  lines?: number;
+  head?: number;
+  noColor: boolean;
+  noHeader?: boolean;
+}
+
+/**
+ * Prints log file content to console with optional filtering
+ */
+async function printLogContent(logPath: string, options: PrintOptions): Promise<void> {
+  const stats = fs.statSync(logPath);
+
+  if (stats.size === 0) {
+    logger.info(chalk.gray('Log file is empty.'));
+    return;
+  }
+
+  // Warn about large files
+  const ONE_MB = 1024 * 1024;
+  if (stats.size > ONE_MB && !options.lines && !options.head) {
+    logger.warn(dedent`
+      Log file is large (${formatFileSize(stats.size)}).
+      Consider using ${chalk.cyan('-n <lines>')} to limit output.
+    `);
+  }
+
+  const content = fs.readFileSync(logPath, 'utf-8');
```

Comment:
This could get ugly if the file is big enough.  Should we use a file stream and and an async iterator?